src/posthog-core.ts

```diff
@@ -387,7 +436,9 @@ export class PostHog {
                 this.persistence.set_user_state('identified')
             }
 
-            config.segment.register(this.segmentIntegration())
+            config.segment.register(this.segmentIntegration()).then(updateInitComplete('segmentRegister'))
```

Comment:
Ah, and we can use `then` here because `config.segment` is bringing its own promise polyfill?

Reply:
Yes exactly. It does [this](https://github.com/segmentio/analytics-next/blob/d0009f839e05deeae7de599c6fdafe0cce84438c/packages/browser/src/browser/standalone.ts#L78).

It would be cool to do something similar as a followup if that's considered the way to do it. It would be _soo_ nice to use modern JS.

Reply:
That would be super cool! 

Looking at https://app.posthog.com/dashboard/65748 and https://github.com/segmentio/analytics-next/blob/d0009f839e05deeae7de599c6fdafe0cce84438c/packages/browser/src/lib/browser-polyfill.ts

This might be a great way to reduce our bundle size for a tonne of visitors